src/sql-parser/src/ident.rs

```diff
@@ -64,8 +76,32 @@ macro_rules! ident {
             }
```

Comment:
nit: while we're here, can you change the `@internal_check_len` to `@internal_checks` since we're no longer checking just the length?